README.md

```diff
@@ -39,6 +39,31 @@ You can also build a debian package and install it:
 * See [SONiC telemetry in dial-out mode](./doc/dialout.md) for how to run dial-out mode system telemetry client
 * See [gNMI Usage Examples](./doc/gNMI_usage_examples.md) for gNMI client usage examples.
 
+### Streaming events
+SONiC publishes events of interest via gNMI.
+Sample events could be bgp-state change, link-state change.
+A sample query URL is as below.
+    `gnmi_cli -client_types=gnmi -a 127.0.0.1:50051 -t EVENTS -logtostderr -insecure -v 7 -streaming_type ON_CHANGE -q all -qt s`
+
+The events URL can take the following optional parameters.<br/>
+heartbeat:<br/>
+    `[heartbeat=<N>]`
+<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;The SONiC switch publishes periodic hearbeats when there are no events to publish.<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;The frequency of the heartbeat can be controlled by this parameter as beat in every N seconds.<br/>
+
+usecache:<br/>
+    `[usecache=true/false]`
+<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;The SONiC switch does offline cache when gNMI client is down.<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;The cached events are delivered upon nest gNMI connection<br/>
+&nbsp;&nbsp;&nbsp;&nbsp;Only one client should use the cache service<br/>
```

Comment:
How to guarantee this limitation?
If two clients use the cache service, what will happen?

Reply:
Limitation by design. 
Trade off between code complexity in supporting multiple clients with cache vs practical use case.
In practice, there is only one client streaming out.
This is added to enable test tools to function w/o disturbing the main service.
To answer your question: If multiple use cache, of course first come first served. The cache will go to one or other.